From bc0b5263fa10eb31aecc69161685f919294cf3c0 Mon Sep 17 00:00:00 2001
From: Neal Gompa <ngompa@fedoraproject.org>
Date: Thu, 9 Oct 2025 10:33:03 -0400
Subject: [PATCH 1001/1003] cargo: Use internal copy of rust-pcre2 instead of
 fetching from git

We separately download and incorporate the forked crate as a
workspace member to allow us to use packaged crates everywhere else.
---
 Cargo.toml | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Cargo.toml b/Cargo.toml
index 100804797..0c433f922 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -1,5 +1,5 @@
 [workspace]
-members = ["crates/*"]
+members = ["crates/*", "third-party-forks/rust-pcre2", "third-party-forks/rust-pcre2/pcre2-sys"]
 
 [workspace.package]
 # To build revisions that use Corrosion (those before 2024-01), use CMake 3.19, Rustc 1.78 and Rustup 1.27.
@@ -53,7 +53,7 @@ nix = { version = "0.31.1", default-features = false, features = [
 ] }
 num-traits = "0.2.19"
 once_cell = "1.19.0"
-pcre2 = { git = "https://github.com/fish-shell/rust-pcre2", tag = "0.2.9-utf32", default-features = false, features = [
+pcre2 = { path = "./third-party-forks/rust-pcre2", default-features = false, features = [
     "utf32",
 ] }
 phf = { version = "0.13", default-features = false }
-- 
2.53.0

